### PR TITLE
[KBFS-1516] Move MdID into kbfsmd package

### PR DESCRIPTION
**Pull request by akalin-keybase**
This removes another server dependency on libkbfs.

Rename it to kbfsmd.ID.

Also remove Crypto.MakeMdID in favor of kbfsmd.MakeID.

Also remove CryptoMeasured.

### CODE REVIEW

File: libkbfs/bare_root_metadata_v2_test.go [hunk 1]
```diff
@@ -299,7 +296,7 @@ func makeFakeBareRootMetadataV2Future(t *testing.T) *bareRootMetadataV2Future {
 				"uid1",
 				0xb,
 				5,
-				MdID{h},
+				tlf.FakeMdID(1),
```
Comment (akalin-keybase) [inline libkbfs/bare_root_metadata_v2_test.go]: Use FakeMdID here

File: libkbfs/crypto_common.go [hunk 2]
```diff
@@ -53,26 +53,6 @@ func (c CryptoCommon) MakeRandomBranchID() (BranchID, error) {
 	return id, nil
 }
 
-// MakeMdID implements the Crypto interface for CryptoCommon.
-func (c CryptoCommon) MakeMdID(md BareRootMetadata) (MdID, error) {
```
Comment (akalin-keybase) [inline libkbfs/crypto_common.go]: Removed this in favor of tlf.MakeMdID

File: libkbfs/errors.go [hunk 3]
```diff
@@ -418,18 +418,6 @@ func (e BadMDError) Error() string {
 	return fmt.Sprintf("Wrong format for metadata for directory %v", e.ID)
 }
 
-// MDMissingDataError indicates that we are trying to take get the
-// metadata ID of a MD object with no serialized data field.
-type MDMissingDataError struct {
```
Comment (akalin-keybase) [inline libkbfs/errors.go]: Move this to tlf

File: libkbfs/kbfs_ops_test.go [hunk 4]
```diff
@@ -75,16 +75,6 @@ func kbfsOpsInit(t *testing.T, changeMd bool) (mockCtrl *gomock.Controller,
 	config.mockBcache = nil
 	config.mockDirtyBcache = nil
 
-	if changeMd {
```
Comment (akalin-keybase) [inline libkbfs/kbfs_ops_test.go]: Removing this code didn't cause any tests to fail, not even with mock errors. So I think it's obsolete now.

File: libkbfs/kbfs_ops_test.go [hunk 5]
```diff
@@ -54,7 +54,7 @@ const (
 // Time out individual tests after 10 seconds.
 var individualTestTimeout = 10 * time.Second
 
-func kbfsOpsInit(t *testing.T, changeMd bool) (mockCtrl *gomock.Controller,
```
Comment (akalin-keybase) [inline libkbfs/kbfs_ops_test.go]: Removed now-unneeded param.

File: tlf/md_id_test.go [hunk 6]
```diff
@@ -2,69 +2,47 @@
...
 	"testing"
 
 	"github.com/keybase/kbfs/kbfscodec"
 	"github.com/keybase/kbfs/kbfshash"
+	"github.com/stretchr/testify/require"
 )
 
 // Make sure MdID encodes and decodes properly with minimal overhead.
 func TestMdIDEncodeDecode(t *testing.T) {
+	id := FakeMdID(1)
```
Comment (akalin-keybase) [inline tlf/md_id_test.go]: Use require module in this file, and FakeMdID.

**Comment by akalin-keybase** [comment]
This is 90% automated renaming. I commented the places which aren't.

File: tlf/bare_root_metadata.go [hunk 7]
```diff
@@ -0,0 +1,16 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package tlf
```
Comment (jzila) [inline tlf/bare_root_metadata.go]: `kbfsmd`

File: tlf/errors.go [hunk 8]
```diff
@@ -29,3 +31,15 @@ func (e HandleExtensionMismatchError) Error() string {
 	return fmt.Sprintf("Folder handle extension mismatch, "+
 		"expected: %s, actual: %s", e.Expected, e.Actual)
 }
+
+// MDMissingDataError indicates that we are trying to take get the
+// metadata ID of a MD object with no serialized data field.
```
Comment (jzila) [inline tlf/errors.go]: `kbfsmd`

File: tlf/md_id.go [hunk 9]
```diff
@@ -2,12 +2,14 @@
 // Use of this source code is governed by a BSD
 // license that can be found in the LICENSE file.
 
-package libkbfs
+package tlf
```
Comment (jzila) [inline tlf/md_id.go]: `kbfsmd`

**Pull request merged**